### DISCUSSION TITLE
use device-detector instead of detect-browser

### DISCUSSION THREAD

**Discussion by ariaieboy** [umami-software/umami#1662] [General]
as you may know, browsers are dropping user-agent string support slightly and right now user-agent does not reflect os versions higher than 10. and we have an issue about it here https://github.com/umami-software/umami/issues/720
right now umami uses https://github.com/DamonOehlman/detect-browser which is an old library that was updated in 2021. 
https://github.com/sanchezzzhak/node-device-detector uses a much bigger database and have better detection it's updating more frequently and it's support [client-hints](https://web.dev/user-agent-client-hints/)

**Comment by rohandebsarkar** [comment]
Another option could be to use something like [ua-parser-js](https://github.com/faisalman/ua-parser-js), which has a much bigger user base.
